base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/pythoninterface/PythonInterface.java

```diff
@@ -107,13 +125,46 @@ public PythonInterface(String actionLoaderPythonScript) {
 				if (!pythonProcess.isAlive()) {
 					break;
 				}
-				LOG.error("> " + line);
+				if (line.equals("Python Ready")) {
```

Comment:
This one could be misleading as we are not really listening for errors on this one even though we are reading from stderr. Could this be separated from this method?

Reply:
I agree that this is not the ideal way to do this, I had planned to replace it once I had a better idea but never found that better idea. We should discuss it with @DominicOram if possible.

Reply:
Have refactored it out to attempt to reconnect with Python in a loop. I have looked to ensure that this does not hang the main gui (i.e. you can still do scripting or looking at spectra plots etc.) but please also check that you're happy it does not as part of the rereview.